rebasehelper/argument_parser.py

```diff
@@ -51,10 +51,12 @@ def __init__(self, option_strings,
                  switch=False,
                  counter=False,
                  append=False,
+                 allow_empty=False,
```

Comment:
This works great, but does `allow_empty` make sense now? Shouldn't combination of `nargs` and `const` be enough?

Reply:
For `pkgcomparetool`, `nargs` and `const` would suffice but what about other options with choices? Without `allow_empty` `--color=""` would be successfully parsed which doesn't make much sense to me.

Also thinking about it now, we should perhaps add `nargs`, `const` and `allow_empty` to some other options (e.g `--versioneer-blacklist`). The option is empty by-default but we should allow users to explicitly specify it...

Reply:
> Without `allow_empty` `--color=""` would be successfully parsed which doesn't make much sense to me.

Why would it be, with no `nargs`?

> Also thinking about it now, we should perhaps add `nargs`, `const` and `allow_empty` to some other options (e.g `--versioneer-blacklist`).

Sure, why not.

Reply:
> Why would it be, with no `nargs`?

Not sure if I understand what you mean by this. With the changes in this PR, if `--pkgcomparetool` is used without additional argument, its value is set to `const` (so `['']`). If an additional argument is used, even if it is an empty string, it is normally read (and due to `type: lambda s: s.split(',')` it's read as `['']`).

So we have to check for empty strings but as I have pointed out in the previous comment, we don't want to enable empty strings by default, it doesn't make sense with some options. `--color` uses choices and its nargs are set to 1 AFAIK.

Reply:
> `--color` uses choices and its nargs are set to 1 AFAIK.

Exactly, so it's not `?`, which means it's not optional. `allow_empty` is equivalent to `nargs == '?'`.